server/src/main/java/org/apache/iotdb/db/engine/compaction/performer/impl/ReadPointCompactionPerformer.java

```diff
@@ -239,6 +238,60 @@ private void compactNonAlignedSeries(
     compactionWriter.endChunkGroup();
   }
 
+  private ConcurrentHashMap<String, MeasurementSchema> getMeasurementSchema(
+      String device, Set<String> measurements) throws IllegalPathException, IOException {
+    ConcurrentHashMap<String, MeasurementSchema> schemaMap = new ConcurrentHashMap<>();
+    List<TsFileResource> allResources = new LinkedList<>(seqFiles);
+    allResources.addAll(unseqFiles);
+    for (String measurement : measurements) {
+      for (TsFileResource tsFileResource : allResources) {
+        if (!tsFileResource.mayContainsDevice(device)) {
+          continue;
+        }
+        MeasurementSchema schema =
+            getMeasurementSchemaFromReader(
+                readerCacheMap.computeIfAbsent(
+                    tsFileResource,
+                    x -> {
+                      try {
+                        return new TsFileSequenceReader(x.getTsFilePath());
+                      } catch (IOException e) {
+                        throw new RuntimeException(
+                            String.format(
+                                "Failed to construct sequence reader for %s", tsFileResource));
+                      }
+                    }),
```

Comment:
Throwing a runtime exception is not a very delicate way. You may catch it using an AtomicReference or something else and throw it outside.